test/jit/test_pdt.py

```diff
@@ -0,0 +1,182 @@
+        self.assertEqual(scripted_fn_bool((True, True, True)),
+                         test_list_and_tuple((True, True, True)))
+
+        scripted_fn_int = torch.jit._script_pdt(test_list_and_tuple, example_inputs=[((3, 4, 5), )])
+        self.assertEqual(scripted_fn_int((1, 2, 3)), test_list_and_tuple((1, 2, 3)))
+
+    def test_pdt_dict(self):
+        def test_dict(a):
+            return a['foo']
+
+        make_global(test_dict)
+
+        _input = {'foo' : True, 'bar': False}
+        scripted_fn = torch.jit._script_pdt(test_dict, example_inputs=[(_input,)])
+        self.assertEqual(scripted_fn({'foo' : False, 'bar': True}, ), test_dict({'foo' : False, 'bar': True}, ))
+
+    def test_pdt_dict_1(self):
+        def test_dict_int_list(a):
+            return a[1]
+
+        make_global(test_dict_int_list)
+
+        _input = {0 : [True, False], 1: [False, True]}
+        scripted_fn = torch.jit._script_pdt(test_dict_int_list, example_inputs=[(_input,)])
+        self.assertEqual(scripted_fn({0 : [False, False], 1: [True, True]}, ),
+                         test_dict_int_list({0 : [False, False], 1: [True, True]}, ))
+
+    def test_any(self):
+        def test_multiple_types(a):
+            assert not isinstance(a, bool)
+            return a
+
+        def test_multiple_types_2(a):
+            assert a is not None
+            return a
+
+        make_global(test_multiple_types, test_multiple_types_2)
+
+        scripted_fn = torch.jit._script_pdt(test_multiple_types, example_inputs=[(1,), ("abc", ), (8.9,), ([3, 4, 5], )])
+        self.assertEqual(scripted_fn(10), test_multiple_types(10))
+        self.assertEqual(scripted_fn("def"), test_multiple_types("def"))
+        self.assertEqual(scripted_fn(7.89999), test_multiple_types(7.89999))
+        self.assertEqual(scripted_fn([10, 11, 14]), test_multiple_types([10, 11, 14]))
+
+        scripted_fn_2 = torch.jit._script_pdt(test_multiple_types_2, example_inputs=[(1,), ("abc", ), (8.9,),
+                                              ([3, 4, 5],), (True, ), ({"a": True}, ), ])
+        self.assertEqual(scripted_fn_2(10), test_multiple_types_2(10))
+        self.assertEqual(scripted_fn_2("def"), test_multiple_types_2("def"))
+        self.assertEqual(scripted_fn_2(7.89999), test_multiple_types_2(7.89999))
+        self.assertEqual(scripted_fn_2([10, 11, 14]), test_multiple_types_2([10, 11, 14]))
+        self.assertEqual(scripted_fn_2(False), test_multiple_types_2(False))
+        self.assertEqual(scripted_fn_2({"abc" : True, "def": False}), test_multiple_types_2({"abc" : True, "def": False}))
+
+    def test_class_methods(self):
+        class M1:
+            def fn(a):  # noqa: B902
+                return a
+
+        make_global(M1)
+
+        scripted_fn = torch.jit._script_pdt(M1.fn, example_inputs=[(10, )])
+        self.assertEqual(scripted_fn(M1.fn(2)), M1.fn(2))  # type: ignore[arg-type]
```

Comment:
There seems to be quite some lint surppression, what are they about? Is there a good reason to have these lint offenses?

Reply:
Yes. So these throw the below errors:
```
error: Argument 1 to "fn" has incompatible type "int"; expected "M1"  [arg-type]
```
So it's expecting the first argument to be of type class `M1` for `fn`

Reply:
I think this is because first argument of a class method should always be `self`, which must be typed as the class. So it is a real issue.

Lint warnings are generally useful at identifying underlying issues, suppressing them is sort of like removing all smoke detectors in a house : )

Reply:
I agree, but adding the self, I run into runtime error:
I don't think even in torch.jit.script this is not supported. 

Here is an example:

```
class example:
   def fn(self, b: int):
      return b

s = torch.jit.script(example.fn)
print(s(20))
```
Traceback (most recent call last):
  File "example.py", line 15, in <module>
    print(s(20))
RuntimeError: fn() Expected a value of type 'Tensor (inferred)' for argument 'self' but instead found type 'int'.
Inferred 'self' to be of type 'Tensor' because it was not annotated with an explicit type.
Position: 0
Value: 20
Declaration: fn(Tensor self, int b) -> (int)
Cast error details: Unable to cast Python instance of type <class 'int'> to C++ type 'at::Tensor'
```
Even this:
```
s = torch.jit.script(example)
print(s.fn(20))

Traceback (most recent call last):
  File "example.py", line 12, in <module>
    print(s.fn(20))
TypeError: fn() missing 1 required positional argument: 'b'

```